Src/DependencyCollector/Net45.Tests/DependencyTrackingTelemetryModuleHttpTest.cs

```diff
@@ -345,6 +344,196 @@ public void OnBeginOnEndAreNotCalledForAppInsightsUrl()
+            }
+        }
+
+        [TestMethod]
+        [Timeout(500000)]
+        public void TestDependencyCollectorPostRequestsAreCollectedEventSource()
+        {
+            using (var module = new DependencyTrackingTelemetryModule())
+            {
+                module.DisableDiagnosticSourceInstrumentation = true;
+                module.ProfileQueryEndpoint = FakeProfileApiEndpoint;
+                module.Initialize(this.config);
+
+                var url = new Uri(LocalhostUrl);
+                HttpWebRequest request = WebRequest.CreateHttp(url);
+                request.Method = "POST";
+                request.ContentLength = 1;
+
+                using (new LocalServer(LocalhostUrl))
+                {
+                    using (var stream = request.GetRequestStream())
+                    {
+                        stream.Write(new byte[1], 0, 1);
+                    }
+
+                    using (request.GetResponse())
+                    {
+                    }
+                }
+
+                this.ValidateTelemetryForEventSource(this.sentTelemetry.Single(), url, request, true, "200");
+            }
+        }
+
+        [TestMethod]
+        [Timeout(5000)]
+        public void TestHttpRequestsWithQueryStringAreCollectedDiagnosticSource()
+        {
+            using (var module = new DependencyTrackingTelemetryModule())
+            {
+                module.ProfileQueryEndpoint = FakeProfileApiEndpoint;
+                module.Initialize(this.config);
+                Assert.IsTrue(DependencyTableStore.IsDesktopHttpDiagnosticSourceActivated);
+
+                var url = new Uri(LocalhostUrl + "123?q=123");
+                HttpWebRequest request = WebRequest.CreateHttp(url);
+
+                using (new LocalServer(LocalhostUrl))
+                {
+                    using (request.GetResponse())
+                    {
+                    }
+                }
+
+                this.ValidateTelemetryForDiagnosticSource(this.sentTelemetry.Single(), url, request, true, "200");
+            }
+        }
+
+        [TestMethod]
+        [Timeout(5000)]
+        public void TestHttpRequestsWithQueryStringAreCollectedEventSource()
+        {
+            using (var module = new DependencyTrackingTelemetryModule())
+            {
+                module.DisableDiagnosticSourceInstrumentation = true;
+                module.ProfileQueryEndpoint = FakeProfileApiEndpoint;
+                module.Initialize(this.config);
+
+                var url = new Uri(LocalhostUrl + "123?q=123");
+                HttpWebRequest request = WebRequest.CreateHttp(url);
+
+                using (new LocalServer(LocalhostUrl))
+                {
+                    using (request.GetResponse())
+                    {
+                    }
+                }
+
+                this.ValidateTelemetryForEventSource(this.sentTelemetry.Single(), url, request, true, "200");
```

Comment:
nitpick: It would be nice if duplicate code was in helper functions instead of copy/paste duplicated.